Merge bug fixes in release 1.4.368 branch for Nuget update

mregen opened the pull request:
-Merge all bug fixes in release branch for Nuget package update
-ETA March 31st

ThomasNehring committed:
Update BuiltInTests.cs with special case compare Null NodeId (#1609)

A  new unit test which will currently fail, due to incorrect treatment of node ids with numeric identifier 0 in the namespace with index 0. ( "NodeId nodeId = new NodeId (0,0)." ).

Of course such a node id is invalid. It seems to be present, however, in some PLCs under certain situations. We encountered such node ids in the context of a customer ticket. The node ids were read into instance of type "DataValue" using the read service call. The evaluation of the read result then failed due to comparisons like the ones in the unit test.

fixes #1610

dependabot[bot] committed:
Bump Serilog.Expressions from 3.2.1 to 3.3.0 (#1714)

Bumps [Serilog.Expressions](https://github.com/serilog/serilog-expressions) from 3.2.1 to 3.3.0.
- [Release notes](https://github.com/serilog/serilog-expressions/releases)
- [Commits](https://github.com/serilog/serilog-expressions/commits/v3.3.0)

---
updated-dependencies:
- dependency-name: Serilog.Expressions
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mrsuciu committed:
Fixed missing TimeFlowsBackward checks and invalid status code on HistoryRead (#1720)

Fixed missing TimeFlowsBackward checks and invalid status code on HistoryRead

dependabot[bot] committed:
Bump Moq from 4.16.1 to 4.17.2

Bumps [Moq](https://github.com/moq/moq4) from 4.16.1 to 4.17.2.
- [Release notes](https://github.com/moq/moq4/releases)
- [Changelog](https://github.com/moq/moq4/blob/main/CHANGELOG.md)
- [Commits](https://github.com/moq/moq4/compare/v4.16.1...v4.17.2)

---
updated-dependencies:
- dependency-name: Moq
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump Microsoft.Extensions.Logging.Abstractions from 6.0.0 to 6.0.1 (#1728)

Bumps [Microsoft.Extensions.Logging.Abstractions](https://github.com/dotnet/runtime) from 6.0.0 to 6.0.1.
- [Release notes](https://github.com/dotnet/runtime/releases)
- [Commits](https://github.com/dotnet/runtime/compare/v6.0.0...v6.0.1)

---
updated-dependencies:
- dependency-name: Microsoft.Extensions.Logging.Abstractions
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mrsuciu committed:
Merge pull request #1727 from OPCFoundation/dependabot/nuget/Moq-4.17.2

Bump Moq from 4.16.1 to 4.17.2

mregen committed:
Use .NET analyzers instead of Roslynator, fix warnings, no bug fixes (#1729)

-use latests analyzers in .NET instead of Roslynator.
-fix warnings and formatting, no bug fixes!

AlinMoldovean committed:
fix CurrentSubscriptionCount at CloseSession() with active subscriptions

mrsuciu committed:
Merge pull request #1731 from AlinMoldovean/diagnostics/CurrentSubscriptionCount_fix

Fix for ServerDiagnostics.CurrentSubscriptionCount decrement at CloseSession

mregen committed:
Offload .NET 6 tests and code coverage to github actions (#1730)

* add buildandtest.yml with test and coverage build

mregen committed:
fix SByte vs Byte collection (#1740)

mrsuciu committed:
Changes for testing and fixing git bug #1694 (#1747)

* Changes for testing and fixing git bug 1694

* Removed the duplicate server address differing by port number

xky0007 committed:
Fix session recreate TransferSubscriptions bug. (#1746)

* Fixed session recreate TransferSubscriptions bug.

1. When session is recreated and there is no subscription is needed to transfer, it will throw Exception - BadNothingToDo. After adding check the subscriptions count, it won't try to subscribe non-existing subscriptions.

2. When session is recreated and there are subscriptions are needed to transer, it transferred wrong subscriptions. It should transfer old subscriptions in old session, but current code transfer new subscriptions in new session to new session - which will always throw Exception - "A subscription can not be transferred due to missing transfer Id"

* Log update and TransferSubscriptions update.

fixes #1733 and #1734

dependabot[bot] committed:
Bump NUnit from 3.13.2 to 3.13.3 (#1741)

Bumps [NUnit](https://github.com/nunit/nunit) from 3.13.2 to 3.13.3.
- [Release notes](https://github.com/nunit/nunit/releases)
- [Changelog](https://github.com/nunit/nunit/blob/v3.13.3/CHANGES.md)
- [Commits](https://github.com/nunit/nunit/compare/v3.13.2...v3.13.3)

---
updated-dependencies:
- dependency-name: NUnit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mregen committed:
Adapt logging abstractions version to target framework (#1751)

* fix the nuget versions

* revert targetframework

* avoid build warning

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OPCFoundation/UA-.NETStandard?pullRequest=1753) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/OPCFoundation/UA-.NETStandard?pullRequest=1753) before we can accept your contribution.<br/>**5** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: ThomasNehring<br/>:white_check_mark: mrsuciu<br/>:white_check_mark: AlinMoldovean<br/>:white_check_mark: xky0007<br/>:white_check_mark: mregen<br/>:x: dependabot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OPCFoundation/UA-.NETStandard?pullRequest=1753) it.</sub>

TimJoehnk committed:
Expose actual network client connected states in IUaPubSubConnections (#1725)

* Expose actual network client connected states in IUaPubSubConnections

mregen committed:
Update README.md and templates (#1749)

- fix a few badges
- add new template for issues and PR

mregen committed:
Fix for Usertokenpolicy Anonymous, security policy default should be None (#1756)

- Only an issue if the configuration does not specify the security policy uri for anonymous token
- If the config does not specify the security policy, for anonymous user token the security profile Basic256Sha256 was chosen. 
- This setting requires for the client to trust the server certificate if it tries to connect with security None, may cause IOP issues when no security is chosen and encryption is required for the anonymous user token.

mregen committed:
Fix dual license headers (#1763)

Headers incorrectly state that the RCL applies to members, however the OPC Source Deliverables license states RCL is applicable only to corporate members. 

The OPC Foundation Technical Control Board on March 30th decided on the following changes with regard of the license headers:

• Change the dual license in the source files so that “members” are replaced with “corporate members”. 

Co-authored-by: Suciu Mircea Adrian <Mircea-Adrian.Suciu@softing.com>

mregen committed:
Update Nuget license information (#1764)

The OPC Foundation Technical Control Board on March 30th decided on the following changes with regard to the license information in the published Nuget packages:

•	Replace the “Redistributables” license for OPC UA Nuget packages with the Dual License.
•	Change the dual license in the source files so that “members” are replaced with “corporate members”. (see #1763)

Effectively, 
- Nuget packages from libraries provided as samples which only build from source code which is covered by a MIT license, are the directly tagged with the MIT license.
- Nuget packages which build from source code where the headers imply the dual license applies, are tagged with the Dual License  'LICENSE.txt' found in the repository root.

mregen committed:
XmlSerializer in .NET 6 Omits Line Breaks by Default (#1760)

* fix xml output in .NET 6
* unify the way xml writer settings are generated and used, add validation tests
